config/initializers/filter_parameter_logging.rb

```diff
@@ -23,21 +23,13 @@ module ParameterFiltering
   ].freeze
 
   ALLOWED_REGEX = /(^|_)ids?|#{Regexp.union(ALLOWED_ATTRIBUTES)}/.freeze
+  # We have to explicitly exclude integer params because
+  # the lambda can only filter string params.
+  DISALLOWED_INTEGER_PARAMS = [:age, :systolic, :diastolic, :duration_in_days]
   SANITIZED_VALUE = "[FILTERED]".freeze
 
-  # Returns the lamba for attributes that are okay to leave in the logs
-  def self.filter
-    lambda do |key, value|
-      unless key.match(ALLOWED_REGEX)
-        case value
-        when String
-          value.replace(SANITIZED_VALUE)
-        else
-          value.to_s.replace(SANITIZED_VALUE)
-        end
-      end
-    end
-  end
+  # Returns the lambda for attributes that are okay to leave in the logs
+  self.filter = lambda { |key, value| value.replace(SANITIZED_VALUE) unless key.match(ALLOWED_REGEX) }
```

Comment:
The decision tree isn't required anymore because it doesn't need to handle integers anymore? Were there other data types that were being handled by that decision tree?

Reply:
The decision tree wasn't doing anything actually. Only the `String` case was doing the `[FILTERED]` replacement, everything else was being skipped.

Reply:
Interesting. The "Is a string" check looks good. How about hashes? Is the `value.to_s.replace(SANITIZED_VALUE)` replacing sensitive hashes with `FILTERED`? 

Do help me understand how this stuff is working if I'm not getting it 😅

Reply:
The filter parameters API takes a lambda that accepts a `key` and a `value` and expects the lambda to mutate the value in place. It iterates over any hash and calls the lambda on each key value pair.

We were doing `value.to_s.replace(SANITIZED_VALUE)` earlier for non-strings, which just constructed a new string and replaced it. The original value was not mutated. Integers can't be mutated in place unlike strings.

Hope this helps.

https://api.rubyonrails.org/classes/ActionDispatch/Http/FilterParameters.html

Reply:
Ha, makes sense to me now. 🚢